### PR TITLE
feat(PHP): Add PHP code for the chapter_computational_complexity

### PR DESCRIPTION
**Pull request by honghuabin**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: codes/php/chapter_computational_complexity/iteration.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    for ($i = 1; $i <= $n; $i++) {
+        // 循环 j = 1, 2, ..., n-1, n
+        for ($j = 1; $j <= $n; $j++) {
+            $res .= "({$i},{$j}), ";
+        }
+    }
+    return $res;
+}
+
+$n = 5;
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/iteration.php:57]: /* Driver Code */

File: codes/php/chapter_computational_complexity/recursion.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    // 归：返回结果
+    while (!empty($stack)) {
+        // 通过“出栈操作”模拟“归”
+        $res += array_pop($stack);
+    }
+    // res = 1+2+3+...+n
+    return $res;
+}
+
+$n = 5;
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/recursion.php:61]: /* Driver Code */

File: codes/php/chapter_computational_complexity/space_complexity.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if ($n == 0) {
+        return null;
+    }
+    $root = new TreeNode(0);
+    $root->left = buildTree($n - 1);
+    $root->right = buildTree($n - 1);
+    return $root;
+}
+
+$n = 5;
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/space_complexity.php:93]: /* Driver Code */

File: codes/php/chapter_computational_complexity/time_complexity.php [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        // 当元素 1 在数组头部时，达到最佳时间复杂度 O(1)
+        // 当元素 1 在数组尾部时，达到最差时间复杂度 O(n)
+        if ($nums[$i] == 1) {
+            return $i;
+        }
+    }
+    return -1;
+}
+
+$n = 8;
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/time_complexity.php:155]: /* Driver Code */

File: codes/php/chapter_computational_complexity/iteration.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * File: iteration.php
+ * Created Time: 2024-02-27
+ * Author: skating (121826725@qq.com)
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/iteration.php:5]: Recommend using your GitHub username

File: codes/php/chapter_computational_complexity/space_complexity.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ * Author: skating (121826725@qq.com)
+ */
+
+/* 函数 */
+function func() {
+    // 执行某些操作
+    return 0;
+}
+
+const a = 0;    // 常量
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/space_complexity.php:14]: Could we define it in the function `constants`?

File: codes/php/chapter_computational_complexity/space_complexity.php [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    return quadraticRecur($n - 1);
+}
+
+/* 指数阶（建立满二叉树） */
+class TreeNode {
+    public $left;
+    public $right;
+    public $val;
+    public function __construct($val)
+    {
```
Comment (krahets) [inline codes/php/chapter_computational_complexity/space_complexity.php:79]: Please unify the formats of the brackets position

**Comment by krahets** [comment]
> If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:
> 
> * [ ]  This PR represents the translation of a single, complete document, or contains only bug fixes.
> * [ ]  The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.
> 
> If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:
> 
> * [ ]  I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
> * [ ]  I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
> * [ ]  The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

Thanks for the PR! Please read this template message and complete the checklist

File: codes/php/chapter_computational_complexity/space_complexity.php (hunk 6 above)
Comment (honghuabin) [reply codes/php/chapter_computational_complexity/space_complexity.php:14]: In PHP, constants can be defined individually or as attributes of a class, but they cannot be defined in methods.

**Pull request closed without merging**